controllers/controller_test.go

```diff
@@ -155,3 +158,87 @@ func TestCorrectsMisconfiguredWf(t *testing.T) {
 		t.Errorf("expected correctedWf=cronWf, but got diff: %s", diff)
 	}
 }
+
+func TestTaskPhaseMatchesWorkflowPhase(t *testing.T) {
+	testTask, err := createTestTask(ctx)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	wf, err := workflows.WaitForTaskWorkflow(ctx, workflowClient, testTask.GetName(), time.Second)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	testTask, err = taskGenClient.Cli.Get(ctx, testTask.GetName(), v1.GetOptions{})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	tcs := []struct {
+		taskPhase     v1alpha1.TaskPhase
+		workflowPhase argo.WorkflowPhase
+	}{
+		{
+			taskPhase:     v1alpha1.TaskRunning,
+			workflowPhase: argo.WorkflowRunning,
+		},
+		{
+			taskPhase:     v1alpha1.TaskPending,
```

Comment:
Make it clear why it is important to not have the pending phase as the first case in the test table.